### ISSUE TITLE
In certain cases, Range.Intersect() does not return Range("<0.0.0") when no intersection

### ISSUE THREAD

**Issue by KevBelisle** [adamreeve/semver.net#43]
C# code for reproducing :
```
var intersection = new Range("^2").Intersect(new Range("^3"));
// intersection will be Range("<=3.0.0 >3.0.0")
```

My use-case : I need to know if there is an intersection or not between 2 ranges.

**Comment by KevBelisle** [comment]
@adamreeve  Any thoughts on this issue and the linked PR?  I can make any changes you think would be needed.

I'd really like to use your package in our software, but this issue is problematic for us.

**Comment by adamreeve** [comment]
I've released your fix in version 1.2.1, thanks!

**Comment by KevBelisle** [comment]
> I've released your fix in version 1.2.1, thanks!

Awesome! Thanks for your great package!
